konfera/views.py

```diff
@@ -1,14 +1,15 @@
-from django.shortcuts import render
+from django.shortcuts import render, get_object_or_404
 from konfera.models import Event, Sponsor
 
 
-def sponsor_list_view(request):
-    events = Event.objects.all()
-    sponsors = Sponsor.objects.all()
-    context = {'events': events,
+def sponsor_list_view(request, event_id):
+    sponsors = Sponsor.objects.select_related('event') \
+                    .filter(event__id=event_id).order_by('type')
```

Comment:
How about you do the first event and than you select related sponsors? This way we could get rid of one DB query. Please test if it works this way.